viz-lib/src/visualizations/chart/Editor/ChartTypeSelect.jsx

```diff
@@ -1,26 +1,31 @@
-import { map } from "lodash";
+import { filter, includes, map } from "lodash";
 import React, { useMemo } from "react";
+import PropTypes from "prop-types";
 import { Select } from "@/components/visualizations/editor";
 import { visualizationsSettings } from "@/visualizations/visualizationsSettings";
 
-export default function ChartTypeSelect(props) {
-  const chartTypes = useMemo(() => {
-    const result = [
-      { type: "line", name: "Line", icon: "line-chart" },
-      { type: "column", name: "Bar", icon: "bar-chart" },
-      { type: "area", name: "Area", icon: "area-chart" },
-      { type: "pie", name: "Pie", icon: "pie-chart" },
-      { type: "scatter", name: "Scatter", icon: "circle-o" },
-      { type: "bubble", name: "Bubble", icon: "circle-o" },
-      { type: "heatmap", name: "Heatmap", icon: "th" },
-      { type: "box", name: "Box", icon: "square-o" },
-    ];
+const allChartTypes = [
+  { type: "line", name: "Line", icon: "line-chart" },
+  { type: "column", name: "Bar", icon: "bar-chart" },
+  { type: "area", name: "Area", icon: "area-chart" },
+  { type: "pie", name: "Pie", icon: "pie-chart" },
+  { type: "scatter", name: "Scatter", icon: "circle-o" },
+  { type: "bubble", name: "Bubble", icon: "circle-o" },
+  { type: "heatmap", name: "Heatmap", icon: "th" },
+  { type: "box", name: "Box", icon: "square-o" },
+];
 
+export default function ChartTypeSelect({ hiddenChartTypes, ...props }) {
+  const chartTypes = useMemo(() => {
     if (visualizationsSettings.allowCustomJSVisualizations) {
-      result.push({ type: "custom", name: "Custom", icon: "code" });
+      allChartTypes.push({ type: "custom", name: "Custom", icon: "code" });
```

Comment:
Since `allChartTypes` now is a global variable - you cannot modify it because it will cause side-effects. Add this line in the beginning of the function: 

```js
const result = [...allChartTypes];
```

and then use `result` everywhere in this function instead of `allChartTypes`